### PR TITLE
Add reflective boundary condition to NewtonianEuler

### PR DESCRIPTION
**Pull request by yoonso0-0**
## Proposed changes

Current PR is only for DG ghost data; subcell version is being under develop.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/NewtonianEuler/BoundaryConditions/Reflection.hpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace NewtonianEuler::BoundaryConditions {
+/*!
+ * \brief Reflecting boundary conditions for Newtonian hydrodynamics.
+ *
+ * Ghost (exterior) data 'mirrors' interior volume data with respect to the
+ * boundary interface. i.e. reverts the normal component of velocity while using
```
Comment (nilsdeppe) [inline src/Evolution/Systems/NewtonianEuler/BoundaryConditions/Reflection.hpp]: `reverts` -> `reverses`

File: src/Evolution/Systems/NewtonianEuler/BoundaryConditions/Reflection.hpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ * same values for other scalar quantities.
+ *
+ * \f{align*}
+ * v_\text{ghost}^i &= v_\text{int}^i - 2[(v^j-v_m^j)n_j]n^i
```
Comment (nilsdeppe) [inline src/Evolution/Systems/NewtonianEuler/BoundaryConditions/Reflection.hpp]: I think the mesh velocity being in only the second term isn't very intuitive (at least not to me!). Could you add a sentence or two explaining that?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
couple small things, please squash immediately. Thanks for implementing this! :D

File: src/Evolution/Systems/NewtonianEuler/BoundaryConditions/Reflection.hpp (hunk 2 above)
Comment (yoonso0-0) [reply src/Evolution/Systems/NewtonianEuler/BoundaryConditions/Reflection.hpp]: Sure :)

**Comment by yoonso0-0** [comment]
Rebased and squashed  :D

**Review by nilsdeppe — APPROVED** [review]
LGTM! Thanks for clearing up the docs! :)

**Pull request merged**